connectivity-android/src/androidMain/kotlin/dev/jordond/connectivity/internal/AndroidConnectivityProvider.kt

```diff
@@ -49,9 +50,31 @@ internal class AndroidConnectivityProvider(
                 }
             }
 
-            manager.registerNetworkCallback(networkRequest, networkCallback)
+            try {
+                manager.registerNetworkCallback(networkRequest, networkCallback)
 
-            awaitClose { manager.unregisterNetworkCallback(networkCallback) }
+                val initialStatus = manager.initialStatus()
+                trySend(initialStatus)
+
+                awaitCancellation()
+            } finally {
+                manager.unregisterNetworkCallback(networkCallback)
+            }
+        }
+    }
+
+    private fun ConnectivityManager.initialStatus(): Connectivity.Status {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            activeNetwork?.let { network ->
+                getNetworkCapabilities(network)?.let { capabilities ->
```

Comment:
This might be blocking I think? AFAIK, there's a callback version of this that you can use, and it'd be safe to do so since you added a try/finally block.